maestro-studio/web/src/ActionModal.tsx

```diff
@@ -0,0 +1,219 @@
+        />
+      ))}
+    </div>
+  )
+}
+
+// https://github.com/pacocoursey/cmdk/blob/9a808d096d6a9a24c0222bf8b0784fbbc6f184d4/cmdk/src/index.tsx#LL875-L883C2
+function useLazyRef<T>(fn: () => T) {
+  const ref = React.useRef<T>()
+
+  if (ref.current === undefined) {
+    ref.current = fn()
+  }
+
+  return ref as React.MutableRefObject<T>
+}
+
+// https://github.com/pacocoursey/cmdk/blob/9a808d096d6a9a24c0222bf8b0784fbbc6f184d4/cmdk/src/index.tsx#L937
+const useScheduleLayoutEffect = () => {
+  const [s, ss] = React.useState<object>()
+  const fns = useLazyRef(() => new Map<string | number, () => void>())
+
+  useLayoutEffect(() => {
+    fns.current.forEach((f) => f())
+    fns.current = new Map()
+  }, [s, fns])
+
+  return (id: string | number, cb: () => void) => {
+    fns.current.set(id, cb)
+    ss({})
+  }
+}
+
+const useFocused = (examples: CommandExample[]): {
+  focused: CommandExample | null
+  moveFocus: (by: number) => void
+  setFocus: (title: string) => void
+} => {
+  const [userSelectedTitle, setUserSelectedTitle] = useState<string | null>(null)
+  const userSelectedIndex = examples.findIndex(e => e.title === userSelectedTitle)
+  const focusedIndex = userSelectedIndex === -1 ? 0 : userSelectedIndex
+  const focused = examples.length === 0 ? null : examples[focusedIndex]
+  const schedule = useScheduleLayoutEffect()
+
+  const scrollSelectedIntoView = () => {
+    const selected = document.querySelector('.action-row[aria-selected=true]')
+    selected?.scrollIntoView({block: 'nearest'})
+  }
+
+  const moveFocus = (by: number) => {
+    const newFocusedIndex = Math.max(0, Math.min(examples.length - 1, focusedIndex + by))
+    schedule(0, scrollSelectedIntoView)
+    if (focusedIndex === newFocusedIndex) return
+    const newSelected = examples[newFocusedIndex]
+    setUserSelectedTitle(newSelected.title)
+  }
+
+  const setFocus = (title: string) => {
+    setUserSelectedTitle(title)
+  }
+
+  return { focused, moveFocus, setFocus }
+}
+
+export const ActionModal = ({ deviceWidth, deviceHeight, uiElement, onEdit, onRun, onClose }: {
+  deviceWidth: number
+  deviceHeight: number
+  uiElement: UIElement
+  onEdit: (example: CommandExample) => void
+  onRun: (example: CommandExample) => void
+  onClose: () => void
+}) => {
+  const unfilteredExamples = useMemo(() => getCommandExamples(deviceWidth, deviceHeight, uiElement), [uiElement])
+  const [query, setQuery] = useState('')
+  const fuse = useMemo(() => new Fuse(unfilteredExamples, {keys: ['title', 'content']}), [unfilteredExamples])
+  const examples = useMemo(() => query ? fuse.search(query).map(r => r.item): unfilteredExamples, [fuse, unfilteredExamples, query])
+  const { focused, moveFocus, setFocus } = useFocused(examples);
+  useHotkeys('up, down', e => {
+    if (e.code === 'ArrowUp') {
+      moveFocus(-1)
+    } else if (e.code === 'ArrowDown') {
+      moveFocus(1)
+    }
+  }, {preventDefault: true, enableOnFormTags: true})
```

Comment:
Is there a way to add `{preventDefault: true, enableOnFormTags: true}` as options to all `useHotkeys` invocations? Otherwise we could extract it out into an object to avoid having to repeat it for all invocations here.